### ISSUE TITLE
Cannot implement JSON API compliant "related" resource link

### ISSUE THREAD

**Issue by HeroicEric** [AgilionApps/ja_serializer#62]
### TL;DR;

I don't think there is currently any way to specify a "related" resource link that is compliant with the JSON API spec.

### The Problem

The current implementation allows a `link` to be specified for a `has_one` relationship like so:

```elixir
defmodule Blog.CommentView do
  has_one :post
    serializer: Blog.PostView,
    link: :post_link

  def post_link(post, conn) do
    # the post is available here but not the comment
    # cannot build /comments/123/post
  end
end
```

While an `:atom` can be specified to use a custom function link in the example above, the model that is being serialized is not available within the function. Since the model isn't available, it's not possible to return a URL that has the `id` of the model being serialized in it.

According to the JSON API spec, [related resource links](http://jsonapi.org/format/#document-resource-object-related-resource-links):

> a related resource link **MUST NOT** change because its relationship's content changes.

Thanks for you hard work so far :smile:

**Comment by alanpeabody** [comment]
Hi @HeroicEric, thanks for your report.

In your example you have a CommentView serializer, if that is the case the link string/function should receive the comment model in the `post_link` function, not the post. Can you verify this example is accurate? (I recognize this is probably a contrived example, but the real example might be more complicated).

I do think you are on to something (has_one relationship link issues) and I am not completely happy with the relationship logic. It is on my ever expanding to-do list to spend some time really analyzing it and improving it. I would appreciate any suggestions and feedback you or anyone has on how it could be improved.

Thanks!

**Comment by HeroicEric** [comment]
@alanpeabody Woah. My apologies. After getting some sleep, it seems to do exactly what I said it didn't :laughing: 

I do agree that relationship links can be improved. I'll see if I can write up something more detailed but I think there should be support for specifying both the "related" and "self" links for a relationship.
